addons/event/views/event_views.xml

```diff
@@ -480,6 +482,7 @@
            <field name="name">Events</field>
            <field name="type">ir.actions.act_window</field>
            <field name="res_model">event.event</field>
+           <field name="target">main</field>
```

Comment:
This is a trick a find, combined with `return this.do_action('event.action_event_view', target:'main'});` in file event_barcode.js, to let the main "purple" navbar available when click on 'previous' button in barcode-scan in 'multi event' mode...

Otherwise I can do it but must replace code in event_barcode.js with that : 
`_onClickPrevious: function(ev) {
    ev.preventDefault();
    const action = {
        type: 'ir.actions.act_window',
        res_model: 'event.event',
        target:'main'
    };
    if (this.isMultiEvent) {
        return this.do_action({...action,
                name: _t('Events'),
                views: [
                    [false, 'tree'], [false, 'kanban'], [false, 'calendar'],
                    [false, 'form'], [false, 'pivot'], [false, 'graph'],
                ],
                view_mode: 'kanban',
            });
    } else {
        return this.do_action({...action,
            res_id: this.action.context.active_id,
            views: [
                [false, 'form']
            ],
            view_mode: 'form',
        });
    }
},`

What do you think is the best?

Reply:
I checked the "target main" thing and I only found 5 other occurrences in the whole code base.
That's usually an indicator that it's not something very "standard".

I would use your other suggestion, it's more verbose but it also seems more standard/robust.